### PR TITLE
Feature mes 3950 make cat be aware

### PR DESCRIPTION
**Pull request by rossfauldsbjss**
Wired up cat BE screens through to office page.
Fixed display of driving faults (vehicle checks) on debrief and office pages.

This PR is specifically for demo purposes at show and tell, 
there is still some work to do on storing vehicle check comments.

### CODE REVIEW

File: src/pages/back-to-office/cat-b/back-to-office.cat-b.page.html [hunk 1]
```diff
@@ -7,7 +7,7 @@
         <img id='dvsa-logo' src='../../../assets/imgs/back-to-office/logo.png' />
     </div>
     <div class="container">
-        <button ion-button class="mes-neutral-button office-button top-button" navPush="OfficePage" id="continue-to-write-up">
+        <button ion-button class="mes-neutral-button office-button top-button" navPush="OfficeCatBPage" id="continue-to-write-up">
```
Comment (szabikr) [inline src/pages/back-to-office/cat-b/back-to-office.cat-b.page.html:10]: So this should come from the page names constants

File: src/pages/back-to-office/cat-be/back-to-office.cat-be.page.html [hunk 2]
```diff
@@ -7,7 +7,7 @@
         <img id='dvsa-logo' src='../../../assets/imgs/back-to-office/logo.png' />
     </div>
     <div class="container">
-        <button ion-button class="mes-neutral-button office-button top-button" navPush="OfficePage" id="continue-to-write-up">
+        <button ion-button class="mes-neutral-button office-button top-button" navPush="OfficeCatBEPage" id="continue-to-write-up">
```
Comment (szabikr) [inline src/pages/back-to-office/cat-be/back-to-office.cat-be.page.html:10]: Same here

File: src/pages/debrief/cat-be/debrief.cat-be.selector.ts [hunk 3]
```diff
@@ -116,14 +116,14 @@ export const getVehicleCheckDrivingFaults =
...
-      || tellMeDFFault.length > 0) {
+    const totalFaults = showMeDFFault.length + tellMeDFFault.length;
+    if (totalFaults > 0) {
       const competency: CommentedCompetency & MultiFaultAssignableCompetency = {
         comment: vehicleChecks.showMeTellMeComments || '',
         competencyIdentifier: CommentSource.VEHICLE_CHECKS,
         competencyDisplayName: CompetencyDisplayName.SHOW_ME_TELL_ME,
         source: CommentSource.VEHICLE_CHECKS,
-        faultCount: showMeDFFault.length + tellMeDFFault.length,
+        faultCount: totalFaults === 5 ? 4 : totalFaults,
```
Comment (ammarhaiderbjss) [inline src/pages/debrief/cat-be/debrief.cat-be.selector.ts:126]: We already have `getVehicleChecksFaultCountCatBE` in the FaultCountProvider so we should look to refactor this at some point

**Review by ammarhaiderbjss — APPROVED** [review]
👍 Love deleting code

**Review by szabikr — APPROVED** [review]
So this needs some more work, fixing TODOs and moving constants. But for now it can go in like this, assuming the work will be continued on it

**Review by edwardwilson — APPROVED** [review]
Same things as above, looks okay but needs some tweaks. I know this is a work in progress to unblock things and more is to come so happy to merge as is

**Pull request merged**